packages/source-map/tests/sourceMap.spec.ts

```diff
@@ -149,6 +149,114 @@ describe('sourceMap', () => {
 			],
 		]);
 
+		expect([...map.getGeneratedStartEnd(
+			`{{|data?.icon?.toString()}}`.indexOf('|'),
+			`{{data?.icon?.toString()|}}`.indexOf('|')
+		)].map(mapped => mapped.slice(0, 2))).toEqual([
+			[
+				0,
+				`(null as any ? ((null as any ? ((null as any ? (this.data)!.icon : undefined)!.toString : undefined))!() : undefined)`.length,
+			],
+		]);
+	});
+	test('Angular template with overlapping ranges', () => {
+		const map = new SourceMap([
+			{
+				sourceOffsets: [
```

Comment:
I think this is where our misunderstanding comes from. When generating the map, I am putting all of the mappings into a single offsets array. It seems that this is not the intended use of the API, but it's a more performative one - less memory consumption.